[JS] Update nodejs dependencies ahead of schema 1.5 release

paulcam206 opened the pull request:
# Description

Fix up outdated NPM package dependencies. Includes an update to the latest version of `typedoc` and a few attendant `typedoc.json` config file changes.

# How Verified

* local builds (Mac/Linux/Win) and site smoke testing

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6476)

licanhua commented on source/nodejs/adaptivecards-templating/typedoc.json:
```diff
@@ -1,17 +1,16 @@
 	],
 	"theme": "markdown",
-	"stripInternal": true
+	"excludeInternal": true
```
Is it the new option which replaced stripinternal?

almedina-ms replied:
I'm also interested in this topic, looks like stripInternal can modify the value of excludeInternal but some people have issues using stripInternal

http://typedoc.org/guides/options/#excludeinternal

https://github.com/TypeStrong/TypeDoc/issues/1469#:~:text=stripInternal%20is%20a%20TypeScript%20option%20%28https%3A%2F%2Fwww.typescriptlang.org%2Ftsconfig%23stripInternal%29.%20In%200.20%2C,for%20setting%20TypeScript%20options%20via%20TypeDoc%20config%20files.

paulcam206 replied:
for what it's worth, we don't currently publish or otherwise consume these docs. I was just updating the configuration to continue working. we should take a more considered approach to API docs as part of our overall documentation effort (see #6316)

licanhua commented:
For the dependency, do we need to run `npm update`? I saw a lot of vulnerabilities in current dependency
```
d:\repo\AdaptiveCards\source\nodejs>npm update --no-save
...
+ @types/react@16.14.16
added 698 packages from 136 contributors, removed 224 packages, updated 1398 packages and audited 2101 packages in 72.618s

88 packages are looking for funding
  run `npm fund` for details

found 445 vulnerabilities (296 moderate, 149 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

paulcam206 commented:
> For the dependency, do we need to run `npm update`? I saw a lot of vulnerabilities in current dependency
> 
> ```
> d:\repo\AdaptiveCards\source\nodejs>npm update --no-save
> ...
> + @types/react@16.14.16
> added 698 packages from 136 contributors, removed 224 packages, updated 1398 packages and audited 2101 packages in 72.618s
> 
> 88 packages are looking for funding
>   run `npm fund` for details
> 
> found 445 vulnerabilities (296 moderate, 149 high)
>   run `npm audit fix` to fix them, or `npm audit` for details
> ```

looks like these are incorrectly-hoisted dependencies. I'll see if I can get them cleaned up...

licanhua commented:
> > For the dependency, do we need to run `npm update`? I saw a lot of vulnerabilities in current dependency
> > ```
> > d:\repo\AdaptiveCards\source\nodejs>npm update --no-save
> > ...
> > + @types/react@16.14.16
> > added 698 packages from 136 contributors, removed 224 packages, updated 1398 packages and audited 2101 packages in 72.618s
> > 
> > 88 packages are looking for funding
> >   run `npm fund` for details
> > 
> > found 445 vulnerabilities (296 moderate, 149 high)
> >   run `npm audit fix` to fix them, or `npm audit` for details
> > ```
> 
> looks like these are incorrectly-hoisted dependencies. I'll see if I can get them cleaned up...

No worry. I'm working on the 2.10.0 npm version change, and I can do audit in my PR

almedina-ms reviewed (approved):
:shipit:

paulcam206 commented:
> > > For the dependency, do we need to run `npm update`? I saw a lot of vulnerabilities in current dependency
> > > ```
> > > d:\repo\AdaptiveCards\source\nodejs>npm update --no-save
> > > ...
> > > + @types/react@16.14.16
> > > added 698 packages from 136 contributors, removed 224 packages, updated 1398 packages and audited 2101 packages in 72.618s
> > > 
> > > 88 packages are looking for funding
> > >   run `npm fund` for details
> > > 
> > > found 445 vulnerabilities (296 moderate, 149 high)
> > >   run `npm audit fix` to fix them, or `npm audit` for details
> > > ```
> > 
> > 
> > looks like these are incorrectly-hoisted dependencies. I'll see if I can get them cleaned up...
> 
> No worry. I'm working on the 2.10.0 npm version change, and I can do audit in my PR

@jonmill - we should schedule some MQ time to clean stuff up... I filed #6481 to track some of the things I noticed while getting these dependencies updated
